### PR TITLE
release: v5.2.1

### LINKED ISSUE #583: New scopes by ZOOM / Plugin not work!
In relation to the changes introduced in the Moodle <> Zoom integration using OAuth described on the Zoom website (https://developers.zoom.us/docs/integrations/oauth-scopes-overview) it is not possible to use the Zoom plugin .

With the latest version (5.2.0) I get the following errors:

Connection status: Connection failed:The Zoom OAuth configuration is missing these required scopes: meeting:read:admin, meeting:write:admin, user:read:admin

Anyone else with this problem?
Do you know which scopes are the ones that need to be declared?
Is a new version of the ZOOM plugin necessary?

Thank you very much in advance!

### LINKED ISSUE #591: Connection status: Connection failed: The Zoom OAuth configuration is missing these required scopes: meeting:read:admin, meeting:write:admin, user:read:admin
I get the following error message:

Connection status: Connection failed: The Zoom OAuth configuration is missing these required scopes: meeting:read:admin, meeting:write:admin, user:read:admin

The following are my Zoom API scopes
image

image

I'm using 
image

### PR DESCRIPTION
**Pull request by jrchamp**
Several nice fixes, thanks all! This should also allow new integrations which only allow granular scopes to not be completely broken.

Note: Tracking field API support is not yet available in the list of granular scopes.

Fixes #583

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,50 @@ permission is required. You should create a separate Server-to-Server OAuth app
...
+    - report:read:user:admin
+- Faster reports for meetings / webinars (Business accounts and higher)
+    - dashboard:read:list_meeting_participants:admin
+    - dashboard:read:list_meetings:admin
+    - dashboard:read:list_webinar_participants:admin
+    - dashboard:read:list_webinars:admin
+- Allow recordings to be viewed (zoom | viewrecordings)
+    - cloud_recording:read:list_recording_files:admin
+    - cloud_recording:read:recording_settings:admin
+    - user:read:list_recordings:admin
```
Comment (jrchamp) [inline README.md:49]: Contrary to Zoom's documentation, there is no user:read:list_recordings:admin. Instead, cloud_recording:read:list_user_recordings:admin seems to be the correct granular scope.
```suggestion
    - cloud_recording:read:list_user_recordings:admin
    - cloud_recording:read:recording_settings:admin
```

File: classes/webservice.php [hunk 2]
```diff
@@ -1051,6 +1108,8 @@ public function get_user_recordings($userid, $from, $to) {
         ];
 
         try {
+            // Classic: recording:read:admin.
+            // Granular: user:read:list_recordings:admin.
```
Comment (jrchamp) [inline classes/webservice.php:1112]:
```suggestion
            // Granular: cloud_recording:read:list_user_recordings:admin.
```

**Review by karenliulll — APPROVED** [review]
Looks great!

**Comment by MBlondin** [comment]
Just updated to v5.2.1 and still getting the Connection status: Connection failed: The Zoom OAuth configuration is missing these required scopes: meeting:read:invitation:admin, meeting:delete:meeting:admin, meeting:update:meeting:admin, user:read:list_schedulers:admin, user:read:settings:admin error

**Comment by jrchamp** [comment]
Hi @MBlondin,

The updated documentation for granular scopes clarifies that the full set of required scopes is now:
- meeting:read:meeting:admin (Get meeting)
- meeting:read:invitation:admin (Get meeting invitation)
- meeting:delete:meeting:admin (Delete meeting)
- meeting:update:meeting:admin (Update meeting)
- meeting:write:meeting:admin (Create meeting)
- user:read:list_schedulers:admin (List schedulers)
- user:read:settings:admin (Get user settings)
- user:read:user:admin (Get user)

**Pull request merged**